server/utils/youtube.ts

```diff
@@ -8,20 +14,20 @@ if (process.env.YOUTUBE_API_KEY) {
   });
 }
 
-export const mapYoutubeResult = (video: YoutubeResult): YoutubeVideo => {
+export const mapYoutubeResult = (video: YoutubeResult): PlaylistVideo => {
+  const videoId = typeof video.id === 'string' ? video.id : video.id.videoId;
```

Comment:
Why can this be two different things? If they're separate API responses maybe it's better to make them separate interfaces